subcommands/devices/list.go

```diff
@@ -137,40 +139,48 @@ func doList(cmd *cobra.Command, args []string) {
 	}
 	t.AddHeader(cols...)
 
-	var dl *client.DeviceList
-	for {
-		var err error
-		if dl == nil {
-			name_ilike := ""
-			if len(args) == 1 {
-				name_ilike = sqlLikeIfy(args[0])
-			}
-			if len(deviceByFactory) > 0 {
-				deviceNoShared = true
-			} else if len(deviceByGroup) > 0 {
-				subcommands.DieNotNil(fmt.Errorf("A factory is mandatory to filter by group"))
-			}
-			dl, err = api.DeviceList(
-				!deviceNoShared, deviceByTag, deviceByFactory, deviceByGroup, name_ilike, deviceUuid)
-		} else {
-			if dl.Next != nil {
-				dl, err = api.DeviceListCont(*dl.Next)
+	name_ilike := ""
+	if len(args) == 1 {
+		name_ilike = sqlLikeIfy(args[0])
+	}
+	if len(deviceByFactory) > 0 {
+		deviceNoShared = true
+	} else if len(deviceByGroup) > 0 {
+		subcommands.DieNotNil(fmt.Errorf("A factory is mandatory to filter by group"))
+	}
+	dl, err := api.DeviceList(!deviceNoShared, deviceByTag, deviceByFactory, deviceByGroup, name_ilike, deviceUuid, showPage)
+	subcommands.DieNotNil(err)
+	row := make([]interface{}, len(showColumns))
+	for _, device := range dl.Devices {
+		if len(device.TargetName) == 0 {
+			device.TargetName = "???"
+		}
+		for idx, col := range showColumns {
+			col := Columns[col]
+			row[idx] = col.Formatter(&device)
+		}
+		t.AddLine(row...)
+	}
+	t.Print()
+	if dl.Next != nil {
+		fmt.Print("\nNext page of devices can be viewed with: ")
+		found := false
+		for i := 0; i < len(os.Args); i++ {
+			arg := os.Args[i]
+			if len(arg) > 2 && arg[:2] == "-p" {
+				fmt.Printf("-p%d ", showPage+1)
+				found = true
+			} else if len(arg) == 6 && arg[:6] == "--page" {
+				fmt.Printf("-p%d ", showPage+1)
+				found = true
+				i++
 			} else {
-				break
+				fmt.Print(os.Args[i], " ")
 			}
```

Comment:
This is quite elaborate logic, but I'm not sure it gives any added value to the user - they should be able to increment a page number themselves.

I think it should be enough to just say that "There are more devices matching your query.  Please use `--page=<page>` option to see next page(s)."